### PR TITLE
Fix and improve E2E tests

### PR DESCRIPTION
**Pull request by stevesoltys**

### CODE REVIEW

**Comment by grote** [comment]
How does this relate to https://github.com/seedvault-app/seedvault/pull/577 ?

**Comment by stevesoltys** [comment]
@grote it'll likely replace most of it

**Comment by grote** [comment]
Slightly unrelated: we should probably update the status badge in the README.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -31,88 +30,33 @@ jobs:
...
+              ./.github/scripts/run_tests.sh
+
+      - name: Upload test results
         if: always()
         uses: actions/upload-artifact@v3
         with:
-          name: seedvault_test_videos
-          path: seedvault_test_videos/**/*.mp4
+          name: ${{ matrix.emulator_type }}-${{ matrix.android_target }}-results
+          path: seedvault_test_results/**/*
```
Comment (grote) [inline .github/workflows/test.yml:62]: Is this uploading logcat as well?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -31,88 +30,33 @@ jobs:
...
-          script: |
-            ./app/development/scripts/provision_emulator.sh "test" "system-images;android-${{ matrix.android_target }};${{ matrix.emulator_type }};x86_64"
-            echo "Generated AVD snapshot for caching."
-
       - name: Assemble tests
         run: ./gradlew :app:assembleAndroidTest
 
       - name: Run tests
-        uses: reactivecircus/android-emulator-runner@v2
+        uses: Wandalen/wretry.action@v1.3.0
```
Comment (grote) [inline .github/workflows/test.yml:40]: we need to retry 3 times? Are things this flaky? do we know where the flakyness comes from?

**Review by grote — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (stevesoltys) [reply .github/workflows/test.yml:62]: Yeah.

File: .github/workflows/test.yml (hunk 2 above)
Comment (stevesoltys) [reply .github/workflows/test.yml:40]: I'm not sure if we do, just leaving this for now. If we see consistency we can remove it, if not we can fix the flakiness.

**Pull request merged**